### ISSUE TITLE
BbsBls sig test fixtures (valid by various W3C docs) not permitted by current OpenAPI spec

### ISSUE THREAD

**Issue by vdods** [w3c-ccg/vc-api#194]
LinkedDataProof specifies only a proof accepting `jws` as the signature https://github.com/w3c-ccg/vc-http-api/blob/main/docs/components/LinkedDataProof.yml :
```
    LinkedDataProof:
      type: object
      description: A JSON-LD Linked Data proof.
      properties:
        type:
          type: string
          description: Linked Data Signature Suite used to produce proof.
        created:
          type: string
          description: Date the proof was created.
        verificationMethod:
          type: string
          description: Verification Method used to verify proof.
        proofPurpose:
          type: string
          description: The purpose of the proof to be used with verificationMethod.
        jws:
          type: string
          description: Detached JSON Web Signature
```

but the various test fixtures allow for other schemas, e.g. case-18.json https://github.com/w3c-ccg/vc-http-api/blob/main/packages/vc-http-api-test-server/__fixtures__/verifiableCredentials/case-18.json :
```
  "proof": {
    "type": "BbsBlsSignature2020",
    "created": "2021-02-23T19:36:07Z",
    "proofPurpose": "assertionMethod",
    "proofValue": "qSjCNJzoDV3hv3gBPoUNN9m5lj8saDBBxC0iDHuFTXXz4PbbUhecmn/L3rPoGuySNatqC4I8VE22xQy0RAowIxoZCC+B2mZQIAb+/JGlXeAlWgEQc71WipfvsfqSn+KmR/rN1FREOy3rtSltyQ92rA==",
    "verificationMethod": "did:key:zUC724vuGvHpnCGFG1qqpXb81SiBLu3KLSqVzenwEZNPoY35i2Bscb8DLaVwHvRFs6F2NkNNXRcPWvqnPDUd9ukdjLkjZd3u9zzL4wDZDUpkPAatLDGLEYVo8kkAzuAKJQMr7N2#zUC724vuGvHpnCGFG1qqpXb81SiBLu3KLSqVzenwEZNPoY35i2Bscb8DLaVwHvRFs6F2NkNNXRcPWvqnPDUd9ukdjLkjZd3u9zzL4wDZDUpkPAatLDGLEYVo8kkAzuAKJQMr7N2"
  }
```

case-19.json (values truncated for brevity) https://github.com/w3c-ccg/vc-http-api/blob/main/packages/vc-http-api-test-server/__fixtures__/verifiableCredentials/case-19.json :
```
  "proof": {
    "type": "BbsBlsSignatureProof2020",
    "created": "2021-02-23T19:37:24Z",
    "nonce": "lEixQKDQvRecCifKl789TQj+Ii6YWDLSwn3AxR0VpPJ1QV5htod/0VCchVf1zVM0y2E=",
    "proofPurpose": "assertionMethod",
    "proofValue": "AAwP/4n.....c1IJp",
    "verificationMethod": "did:key:zUC724vuG...7N2#zUC724v...7N2"
  }
```

**Comment by msporny** [comment]
The group discussed this on the 2022-04-12 telecon. Someone needs to take an action to look at this and ensure we support the new Data Integrity work, the legacy jws work, and the new VC-JWT work.

**Comment by msporny** [comment]
The group discussed this topic again on the 2023-06-20 telecon:

It was noted that things have moved on from where the issue was a while ago. Namely, we now have endpoint instances, which enable us to have different proof formats and could theoretically issue any Data Integrity Proof format, or AnonCredsv1, or VC-JWT, or SD-JWT, or mdoc, or any other related digital credential format. @PatStLouis noted that testing could be a concern as that might balloon the number of tests we have to write. There was a question from GregB about how much of this comes under media type selection / content negotiation. For example, if someone makes a call to /credentials/issue and requests `application/vc+ld+jwt`, then would the endpoint return a VC-JWT instead of a Data Integrity protected credential? @msporny suggested that it requires more than just a media type to create a specific type of securing mechanism. It was suggested that specific endpoints can return specific types of secured data.

The VC API components need to be updated to match the latest Data Integrity work in the VCWG. Specifically, this file needs to be updated: https://github.com/w3c-ccg/vc-api/blob/main/components/LinkedDataProof.yml (and anything that integrates with it).

If implementers would like to see other securing formats supported by the API, the group asks that issues be raised for the specific securing format and details on how they would be implemented be included in the issue.

**Comment by msporny** [comment]
The group discussed this on 2023-12-05 and noted that we just need to align w/ work at W3C VCWG.

**Comment by msporny** [comment]
PR #358 has been raised to address this issue. This issue will be closed once PR #358 has been merged.

**Comment by msporny** [comment]
R #358 has been merged, closing.
